src/components/BeatCollection.tsx

```diff
@@ -103,22 +121,50 @@ const BeatCollectionComponent: React.FC<BeatCollProps> = ({
 
   return (
     <>
-      <div className="mb-6 flex w-full justify-between">
-        <h2 className="text-2xl font-bold pl-0 pb-0">{currentCollection.set_name}</h2>
-        <div className="mt-4 flex">
-          <div className="mr-4">
-            Venue: {currentCollection.venue || 'N/A'}
+      <div className="w-full flex items-center justify-between px-4 min-h-[64px] bg-gray-700 rounded-xl my-4 bg-opacity-80 backdrop-blur-3xl shadow-sm">
+        <h2 className="text-2xl font-bold text-white truncate max-w-[300px] flex items-center m-0 mt-2">
+          {currentCollection.set_name}
+        </h2>
+
+        <div className="flex items-center gap-8 xl:gap-12 text-xl" >
+          {/* Always visible - even on small screens */}
+          <div className="flex items-center whitespace-nowrap">
+            <span className="text-gray-400 mr-2">beats:</span>
+            <span>{collectionBeats.length}</span>
           </div>
-          <div>
-            Date Played: {currentCollection.date_played || 'N/A'}
+
+          {/* Hidden on screens smaller than 1280px */}
+          <div className="hidden xl:flex items-center gap-8">
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2 text">Venue:</span>
+              <span>{currentCollection.venue || 'N/A'}</span>
+            </div>
+
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2">Date:</span>
+              <span>{formatDate(currentCollection?.date_played)}</span>
+            </div>
+
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2">City:</span>
+              <span>{currentCollection.city || 'N/A'}</span>
+            </div>
+
+            <div className="whitespace-nowrap">
+              <span className="text-gray-400 mr-2">State:</span>
+              <span>{currentCollection.state_name || 'N/A'}</span>
+            </div>
```

Comment:
personally I think the text is a little large on the screen for these set attributes. I do really like how they hide on the small screen though.

Might be worthwhile to add total beats to the All Beats screen?